### PR TITLE
Add reconciliation duration parameter and periodic reconciliation test

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Remove leftover comment and env
- Add period startup parameter
- Add test for periodic reconcilation

### CODE REVIEW

File: controllers/apirule_controller.go [hunk 1]
```diff
@@ -60,13 +58,15 @@ type APIRuleReconciler struct {
...
+	ReconcilePeriod        time.Duration
+	OnErrorReconcilePeriod time.Duration
 }
 
 const (
 	CONFIGMAP_NAME               = "api-gateway-config"
 	CONFIGMAP_NS                 = "kyma-system"
 	DEFAULT_RECONCILATION_PERIOD = 30 * time.Minute
-	ERROR_RECONCILATION_PERIOD   = time.Minute
+	ERROR_RECONCILATION_PERIOD   = 5 * time.Minute
```
Comment (barchw) [inline controllers/apirule_controller.go:69]: I'm not sure about those default periods. Do you think this is OK, or should be longer/shorter?
```suggestion
	ERROR_RECONCILATION_PERIOD   = 5 * time.Minute
```

Comment (werdes72) [reply controllers/apirule_controller.go:69]: I would go with something like 1 minute.

File: controllers/api_controlller_reconcilation_test/api_controller_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
+package reconcilation_test
```
Comment (werdes72) [inline controllers/api_controlller_reconcilation_test/api_controller_integration_test.go:1]: It should be `reconciliation` AFAIK

File: controllers/api_controlller_reconcilation_test/suite_test.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
+package reconcilation_test
```
Comment (werdes72) [inline controllers/api_controlller_reconcilation_test/suite_test.go:1]: It should be `reconciliation` AFAIK

File: controllers/api_controlller_reconcilation_test/api_controller_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+	kind                        = "APIRule"
+	testGatewayURL              = "kyma-system/kyma-gateway"
+	testOathkeeperSvcURL        = "oathkeeper.kyma-system.svc.cluster.local"
+	testOathkeeperPort   uint32 = 1234
+	testNamespace               = "atgo-system"
+	testNameBase                = "test"
+	testIDLength                = 5
+)
+
+var _ = Describe("APIRule Controller Reconcilation", func() {
```
Comment (werdes72) [inline controllers/api_controlller_reconcilation_test/api_controller_integration_test.go:42]: It should be `reconciliation` AFAIK

File: controllers/apirule_controller.go (hunk 1 above)
Comment (werdes72) [reply controllers/apirule_controller.go:69]: Also typo here `reconciliation`

**Review by werdes72 — APPROVED** [review]

**Pull request merged**